src/Avalonia.Controls.DataGrid/DataGrid.cs

```diff
@@ -2392,6 +2406,9 @@ internal bool UpdateScroll(Vector delta)
                         // We don't need to invalidate once again after UpdateHorizontalOffset.
                         ignoreInvalidate = true;
                         handled = true;
+                        
+                        var eventType = delta.X > 0 ? ScrollEventType.SmallDecrement : ScrollEventType.SmallIncrement;
+                        HorizontalScroll?.Invoke(this, new ScrollEventArgs(eventType, delta.X));
```

Comment:
Shouldn't that use the clamped `horizontalOffset` rather than the original `delta.X`?